lib/migrate.js

```diff
@@ -0,0 +1,505 @@
+
+        async.series([
+          function dropIndexes(cb) {
+            removeIndexes(results.indexesToDrop, cb);
+          },
+          function addIndexes(cb) {
+            createIndexes(results.indexesToAdd, cb);
+          },
+        ], cb);
+      });
+
+      function createIndexes(indexes, cb) {
+        assertMsg = helpers.doAssert(typeof indexes === 'object',
+          'indexes to create must be an object!');
+        if (assertMsg) return cb(assertMsg);
+        async.eachOf(indexes, create, cb);
+        // {indexName: "foo"} or {indexName: [{"foo": "asc"}, {"bar": "asc"}]}
+        function create(value, key, cb) {
+          createIndex(key, value, cb);
+        };
+      };
+
+      function createIndex(name, fields, cb) {
+        fields = self.coerceIndexFields(fields);
+        // naming convertion: '_design/LBModel__Foo__LBIndex__foo_index',
+        // here the driver api takes in the name without prefix '_design/'
+        var config = {
+          ddocName: self.getIndexModelPrefix(mo) + '__' + model + '__' +
+          self.getIndexPropertyPrefix(mo) + '__' + name,
+          indexName: name,
+          fields: fields,
+        };
+        self.createIndex(config.ddocName, config.indexName, config.fields, cb);
+      }
+
+      function removeIndexes(indexes, cb) {
+        assertMsg = helpers.doAssert(typeof indexes === 'object',
+          'indexes to drop must be an object!');
+        if (assertMsg) return cb(assertMsg);
+        async.eachOf(indexes, removeIndex, cb);
+        // {ddoc: "_design/ddocName", name: "indexName"}
+        function removeIndex(value, key, cb) {
+          self.deleteIndex(mo, value.ddoc, cb);
+        }
+      };
+    };
+  };
+
+  /**
+ * Coerce the index fields to be either ALL ASC or ALL DESC if conflict exists.
+ * If the fields are specified with different orders:
+ * - select the order that is on the first field
+ * - force all fields using that order
+ * - print warning for coerced fields' names
+ * 
+ * @param {Array} fields The fields defined in an index, example:
+ * ```js
+ * [{foo: 'asc'}, {bar: 'desc'}]
+ * ```
+ * @returns {Array} The coerced fields
+ */  
+  CouchDB.prototype.coerceIndexFields = function(fields) {
+    if (fields.length <= 1) return fields;
+    var defaultOrder = fields[0][Object.keys(fields[0])[0]];
+    var coercedFields = [];
+
+    var result = _.map(fields, coerceOrder);
+    if (coercedFields.length > 0) {
+      printWarning();
+    }
+    return result;
+
+    function coerceOrder(field) {
+      if (field[Object.keys(field)[0]] === defaultOrder)
+        return field;
+      field[Object.keys(field)[0]] = defaultOrder;
+      coercedFields.push(Object.keys(field)[0]);
+      return field;
+    }
+
+    function printWarning() {
+      var couchRule = 'Couchdb does NOT allow composite indexes go opposite direction, ' +
+        'please see http://docs.couchdb.org/en/2.0.0/api/database/find.html#db-index for details.' + '\n';
```

Comment:
Minor correction: `CouchDB does not allow composite indexes with conflicting sort directions, please see ...`